pkg/prometheus/resource_selector.go

```diff
@@ -827,6 +827,11 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			continue
 		}
 
+		if err = rs.validateStaticConfig(sc); err != nil {
```

Comment:
```suggestion
		// The Kubernetes API can't do the validation (for now) because kubebuilder validation markers don't work on map keys with custom type.
		// https://github.com/prometheus-operator/prometheus-operator/issues/6889
		if err = rs.validateStaticConfig(sc); err != nil {
```